pages/GSoC-21.md

```diff
@@ -0,0 +1,156 @@
+
+**Skills Required**:
+- Python
+- Hydra
+- Front and Backend development skills
+- Google Cloud
+- Semantic Web
+- Ability to write test suites
+
+**Difficulty Level**: *Intermediate* to *Hard*
+
+You can discuss each idea in detail on the Slack channel. We would be happy to provide more info if anyone is interested in any of the above projects.
+
+
+## 4. Make [hydrus](https://github.com/HTTP-APIs/hydrus) future proof
+### Key points:
+Right now hydrus is a generic flask compatible server. Most commercial APIs contain large amounts of data distributed over multiple servers. The idea here is to improve the scalability of the database by deploying a hydrus instance for a set of <em>HydraClass</em>'es and then allowing linking by reference.
+The parser should call for the deployment of a hydrus instance for each set of classes in the APIDoc, predicates between classes should be represented as links; by consequence, the agent should build its own representation by dereferencing the links.
+
+**Skills Required**:
+- Python
+- Hydra
+- Agent
+- hydrus
+- Semantic Web
+- Software Engineering
+
+**Difficulty Level**: *Hard*
+
+**NOTE**: This is a sophisticated project and will require a significant understanding of various HydraEcosystem tools including hydrus, parser, and the agent. We understand there's a lot to be done here, so, we'll try to pair 2-3 students to complete the overall task providing students an opportunity to learn how to work in distributed teams.
+
+## 5. Get Creative
+We encourage creativity a lot and understand it is very important for you to work on the things you're really interested in. If you want, you can choose subparts from the ideas listed above or even come up with your own (discussing new ideas with mentors first is highly encouraged) to create the proposal best suited to your interests.
+
+### Skills Required:
+- A TON OF CREATIVITY
+
+----
+
+
+# [How do I get started ?](#getting-started)
+ Getting started is pretty easy. Head over to our [homepage](http://hydraecosystem.org/) or the W3C [community page](https://www.hydra-cg.com/). There are a lot of demos, presentations, and talks to get you up to speed. Then head over to the [hydrus repo](https://github.com/HTTP-APIs/hydrus) and clone it. Play with it a little, try to understand how the current implementation works, try to fix some bugs or report any issues you can find [here](https://github.com/HTTP-APIs/hydrus/issues).
+Lastly, don't hesitate to reach out in the [Slack community](https://join.slack.com/t/hydraecosystem/shared_invite/enQtNzM3NTg5NzQ2MDUxLWU1MjM3ZGRhZWM4ZTg1ODBjMTljNTQwNzAwMGM3ZDlmYTY3Y2E4OGJmN2NlZWRjMWIzY2MzN2NjOTIyYmQ1ZjU) if you have any question, we are very friendly people and we'll be more than happy to help you out.
```

Comment:
https://join.slack.com/t/hydraecosystem/shared_invite/zt-ly4j3hg3-gVhJxgc6ykpGrel_N4Xvvg Update the link